feat: dependabot.yaml template

marek-saji opened the pull request:
# Why?

Template to promote more consistent config across projects.

Will follow with a PR adding support for this in [@verkstedt/setup-repo](https://www.npmjs.com/package/@verkstedt/setup-repo).

Closes https://verkstedt.atlassian.net/browse/VIP-98

marek-saji committed:
feat: dependabot.yaml template

marek-saji commented on templates/dependabot.yaml:
```diff
@@ -0,0 +1,29 @@
+
+version: 2
+updates:
+  - package-ecosystem: "npm"
+    directory: "/"
+    schedule:
+      interval: "quarterly"
+      time: "12:00"
+      timezone: "Europe/Berlin"
+    open-pull-requests-limit: 5
+    cooldown:
+      default-days: 7
```
We have security updates enabled in repo settings, so dependabot should create thesese PRs independently from non–security updates.

smelchior commented on templates/dependabot.yaml:
```diff
@@ -0,0 +1,32 @@
+# This file has been created from
+# https://github.com/verkstedt/.github/blob/main/templates/dependabot.yaml
+#
+# IMPORTANT: Dependabot should be configured to create PRs for security
```
We should define what settings should/must be enabled

marek-saji replied:
It’s set up on the organisation level — https://github.com/organizations/verkstedt/settings/security_products/configurations/edit/232473 (+ enabled “Grouped security updates” in https://github.com/organizations/verkstedt/settings/security_analysis)

If you want to get into more detailed conversation about that, let’s do it privately on slack.

Maybe I could change this comment to?:

```yaml
# IMPORTANT: This does not affect creation of PRs with security updates.
# That is configured on GitHub settings and should be done on the
# organization level, if possible.
```

smelchior replied:
Thats fine with me, comments looks like a good idea :)

marek-saji replied:
Updated in https://github.com/verkstedt/.github/commit/70bee328b2ea33a76cac7f03ecb12f6a5986c7c9

smelchior commented on templates/dependabot.yaml:
```diff
@@ -0,0 +1,32 @@
+  - package-ecosystem: "npm"
+    directory: "/"
+    schedule:
+      interval: "quarterly"
```
Does this also affect the security Updates?

marek-saji replied:
No. See previous comment. :slightly_smiling_face:

marek-saji committed:
fixup! feat: dependabot.yaml template

Improve comment about security updates being not affected by this
config.

Resolves https://github.com/verkstedt/.github/pull/5#discussion_r2576731704
